### PR TITLE
Optimization RegexRunner.cs

### PR DESCRIPTION
**Pull request by Sin333**
Hi I'm finded something strange operation please look this.
It's optimization for RegexRunner.cs.

-Delete duplicate multiplication operation
-Remove unused value (int remaining)

### CODE REVIEW

File: src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs [hunk 1]
```diff
@@ -299,8 +299,9 @@ private void InitMatch()
 
             InitTrackCount();
 
-            int tracksize = runtrackcount * 8;
-            int stacksize = runtrackcount * 8;
+            int tmpmultiplicationsize = runtrackcount * 8;
+            int tracksize = tmpmultiplicationsize;
+            int stacksize = tmpmultiplicationsize;
```
Comment (stephentoub) [inline src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs]: The tmpmultiplicationsize isn't necessary.  You can just change it to:
```C#
int tracksize = runtrackcount * 8;
int stacksize = tracksize;
```

File: src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs [hunk 2]
```diff
@@ -345,9 +346,10 @@ private Match TidyMatch(bool quick)
         /// </summary>
         protected void EnsureStorage()
         {
-            if (runstackpos < runtrackcount * 4)
+            int tmpmultiplicationsize = runtrackcount * 4;
+            if (runstackpos < tmpmultiplicationsize)
                 DoubleStack();
-            if (runtrackpos < runtrackcount * 4)
+            if (runtrackpos < tmpmultiplicationsize)
```
Comment (stephentoub) [inline src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs]: I don't think this is a worthwhile change.  It's not going to improve the generated asm, and it makes the code harder to read.  Let's let the compiler do its thing.

**Comment by stephentoub** [comment]
cc: @ViktorHofer

**Comment by danmosemsft** [comment]
@Sin333 thanks for the attempt though! If you want to take another run at improving regex perf, you'll want to start by getting set up to run our microbenchmarks against your local changes. There are instructions in the repo https://github.com/dotnet/performance and the specific tests are here https://github.com/dotnet/performance/tree/master/src/benchmarks/micro/corefx/System.Text.RegularExpressions . If you can improve regex perf in a way measurable by such benchmarks (or similar) and the code change is otherwise reasonable (eg., does not make things unnecessarily hard to maintain) we would be interested.

File: src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs (hunk 2 above)
Comment (Sin333) [reply src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs]: Okey, you're right.
I just wanted to help improve the Regex class.

**Comment by danmosemsft** [comment]
I'm going to close as per @stephentoub comments above. Thanks again

File: src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexRunner.cs]: And we appreciate that.  Thanks for your attempt.

**Pull request closed without merging**